src/frontend/Komponenter/Oppgavebenk/OppgaveKnapp.tsx

```diff
@@ -0,0 +1,139 @@
+    oppdaterOppgave,
+}) => {
+    const {
+        gåTilBehandleSakOppgave,
+        gåTilJournalføring,
+        laster,
+        plukkOppgaveOgGåTilBehandlingsoversikt,
+        tilbakestillFordeling,
+        settOppgaveTilSaksbehandler,
+    } = useOppgave(oppgave);
+
+    const { innloggetSaksbehandler } = useApp();
+
+    const utførHandling = () => {
+        switch (utledHandling(oppgave)) {
+            case Handling.JOURNALFØR:
+                gåTilJournalføring('stønad');
+                break;
+            case Handling.JOURNALFØR_KLAGE:
+                gåTilJournalføring('klage');
+                break;
+            case Handling.SAKSBEHANDLE:
+                gåTilBehandleSakOppgave();
+                break;
+            default:
+                plukkOppgaveOgGåTilBehandlingsoversikt(utledetFolkeregisterIdent(oppgave));
+        }
+    };
+
+    const utførHandlingOgOppdaterOppgave = (handling: () => Promise<void>) => async () => {
+        await handling();
+        oppdaterOppgave();
+    };
+
+    const oppgaveTilordnetInnloggetSaksbehandler =
+        oppgave.tilordnetRessurs === innloggetSaksbehandler.navIdent;
+
+    const skalViseFortsettKnapp =
+        oppgaveTilordnetInnloggetSaksbehandler &&
+        [Handling.SAKSBEHANDLE, Handling.JOURNALFØR, Handling.JOURNALFØR_KLAGE].includes(
+            utledHandling(oppgave)
+        );
+
+    if (skalViseFortsettKnapp) {
+        return (
+            <FlexContainer>
+                <TabellKnapp
+                    type={'button'}
+                    variant={'secondary'}
+                    size={'small'}
+                    onClick={utførHandling}
+                    disabled={laster}
+                >
+                    Fortsett
+                </TabellKnapp>
+                <OppgaveValgMeny
+                    valg={[
+                        {
+                            label: 'Fjern meg',
+                            onClick: utførHandlingOgOppdaterOppgave(tilbakestillFordeling),
+                        },
+                    ]}
+                />
+            </FlexContainer>
+        );
+    } else if (oppgave.tilordnetRessurs) {
+        return (
+            <FlexContainer>
+                {oppgave.tilordnetRessurs}
+                <OppgaveValgMeny
+                    valg={[
+                        {
+                            label: 'Overta',
+                            onClick: utførHandlingOgOppdaterOppgave(settOppgaveTilSaksbehandler),
+                        },
+                    ]}
+                />
+            </FlexContainer>
+        );
+    } else
+        return (
+            <TabellKnapp
+                type={'button'}
+                variant={'secondary'}
+                size={'small'}
+                onClick={utførHandling}
+                disabled={laster}
+            >
+                Tildel meg
+            </TabellKnapp>
+        );
+};
+
+type oppgaveValg = { label: string; onClick: () => void };
```

Comment:
UpperCamelCase for typer :-)
```suggestion
type OppgaveValg = { label: string; onClick: () => void };
```